### DISCUSSION TITLE
Using ReplaceAll in FeynCalc

### DISCUSSION THREAD

**Discussion by sunnyasquabla** [FeynCalc/feyncalc#246] [Q&A] [answered]
Hi,
I am trying to simply an expression for which I have collected Epsilon(p1) . Epsilon(p2) as shown below.
![Screenshot at 2024-02-07 16-33-50](https://github.com/FeynCalc/feyncalc/assets/107980747/9bf34d82-0dde-45cc-91eb-e864769ab59c)

Now I want to take these Epsilon(p1) . Epsilon(p2) term to zero to extract the relevant part that I need.
For this I use ReplaceAll, but it takes every epsilon to zero.
![Screenshot at 2024-02-07 16-38-26](https://github.com/FeynCalc/feyncalc/assets/107980747/f5e4e2b6-3d8e-48a3-93e3-c12e26a75b0b)

I want to keep terms like (p1. Epsilon(p2)) (p1Epsilon(p1)) for further calculations.
Can someone tell me how to do that?
I'll be very thankful to you.

**Comment by vsht** [comment]
Hi, 

just use something like
```
ReplaceAll[#, 
  Pair[Momentum[Polarization[p1, __]], 
    Momentum[Polarization[p2, __]]] :> 0] &
```
and so on/

**Reply by sunnyasquabla** [reply, marked as answer]
Hey,

The above command didn't work
But this one worked,

`ReplaceAll[#, 
 Pair[Momentum[Polarization[___], ___], 
   Momentum[Polarization[___], ___], ___] -> 0]`

I want to know, how do I decide where to put the underscores?
Thanks

**Reply by vsht** [reply]
> I want to know, how do I decide where to put the underscores?

This is not a question about FeynCalc but rather how pattern matching works in Mathematica. I'd suggest that you
first familiarize yourself with the concept of replacement rules and patterns, otherwise you will not get far trying to use FeynCalc.
Also, [mathematica.stackexchange.com](https://mathematica.stackexchange.com) is probably a better place for such inquiries.

**Reply by sunnyasquabla** [reply]
Alright,
Thank you.
